services/task-svc/internal/task/task.go

```diff
@@ -530,3 +533,19 @@ func (ServiceServer) UnpublishTask(ctx context.Context, req *pb.UnpublishTaskReq
 
 	return &pb.UnpublishTaskResponse{}, nil
 }
+
+func GetTaskByPatient(ctx context.Context, patientId uuid.UUID) ([]Task, error) {
+	db := hwgorm.GetDB(ctx)
+
+	var tasks []Task
+
+	if err := db.Preload("Subtasks").Where("patient_id = ?", patientId).Find(&tasks).Error; err != nil {
+		if hwgorm.IsOurFault(err) {
+			return nil, status.Error(codes.Internal, err.Error())
+		} else {
+			return nil, status.Error(codes.InvalidArgument, "id not found")
+		}
```

Comment:
```suggestion
		return err
```

Just return the error here. The handling should happen near the method.